packages/SwingSet/docs/garbage-collection.md

```diff
@@ -64,46 +135,241 @@ When tracking the reachability state of a JavaScript `Object`, we define five st
 * COLLECTED: the JS engine has realized the object is gone
 * FINALIZED: a `FinalizationRegistry` callback is being run
 
-If you had a `WeakRef` for the object, it would be "full" (i.e. `.deref()` returns a value) in the REACHABLE and UNREACHABLE states, and "empty" (`wr.deref() === undefined`) in COLLECTED and FINALIZED.
+If you had a `WeakRef` for the object, it would be "alive" (i.e. `.deref()` returns a value) in the REACHABLE and UNREACHABLE states, and "dead" (`wr.deref() === undefined`) in COLLECTED and FINALIZED.
 
-Note that there's no actual state machine with those values, and we can't observe all of the transitions from JavaScript, but we can describe what operations could cause a transition, and what our observations allow us to deduce about the state:
+Note that there's no actual state machine with those values, and we can't observe all of the transitions from JavaScript. But we *can* describe what operations could cause a transition, and what our observations allow us to deduce about the state:
 
 * UKNOWN moves to REACHABLE when a delivery introduces a new import
```

Comment:
UNKNOWN also moves to REACHABLE when JS code creates an object locally.  Not part of the machinery you've just implemented but important.

Reply:
True, although only when it's also reachable from some vat root (like when it gets returned at the end of a method, or stored in a global, etc). But that end of its lifetime isn't so interesting... I'll add a few words without getting into the weeds.

Reply:
```
* UKNOWN moves to REACHABLE when a delivery introduces a new import
  * or the vat exports/stores a newly created object
  * userspace holds a reference only in REACHABLE
```